pkgs/development/libraries/glib/schema-override-variable.patch

```diff
@@ -0,0 +1,12 @@
+--- a/gio/gsettingsschema.c
++++ b/gio/gsettingsschema.c
+@@ -352,6 +352,9 @@
+ 
+       try_prepend_data_dir (g_get_user_data_dir ());
+ 
++      if ((path = g_getenv ("NIX_GSETTINGS_OVERRIDES_DIR")) != NULL)
++        try_prepend_dir (path);
++
```

Comment:
If you call it overrides, shouldn't it be prepended last, to have maximal priority? And if these are just additions, maybe call them `NIX_EXTRA_GSETTINGS_DIR` or something like that?

Reply:
I assumed the `GSETTINGS_SCHEMA_DIR` is for debugging, which is why I made kept it the highest priority. I named it overrides since its purpose is to load defaults overridden by the `gnome3` NixOS module.

Reply:
I expect this option to be used to add and not override anything, so I would use something like `NIX_GSETTINGS_SCHEMA_DIR`.

I am not convinced, but I agree it is not worth further bikeshedding. The change looks good in general.

Reply:
Well, glib acts like if it traversed all the directories on the list and added the default values for each key described in the directory to a dictionary, but did not override the keys already in the dictionary – basically the first match is used, hence the override.